src/web-push-lib.js

```diff
@@ -285,7 +287,11 @@ WebPushLib.prototype.sendNotification =
 
       httpsOptions.headers = requestDetails.headers;
       httpsOptions.method = requestDetails.method;
-
+      
+      if('proxy' in options && typeof options.proxy === 'string') {
+          httpsOptions.agent = new HttpsProxyAgent(options.proxy);
```

Comment:
Could you separate the check for the type of `options.proxy` and print a console warning if it's set but not used because it's not a string.

I'd also be happy with throwing an error.